site/docs/spec.md

```diff
@@ -212,6 +212,9 @@ Columns in Iceberg data files are selected by field id. The table schema's colum
 
 For example, a file may be written with schema `1: a int, 2: b string, 3: c double` and read using projection schema `3: measurement, 2: name, 4: a`. This must select file columns `c` (renamed to `measurement`), `b` (now called `name`), and a column of `null` values called `a`; in that order.
 
+Tables may also define a property `schema.name-mapping.default` with a JSON map of `columnName` -> `fieldId` which will be used if a data file was written without field ids. This `NameMapping` will **only** be used on files without field ids. Files imported or added to an Iceberg table from a system that does not generate field ids will fall back to using the table's name mapping to map columns to field ids.
```

Comment:
I'd change "This NameMapping will only" to "This NameMapping may only" because we're not describing behavior, we are setting requirements for behavior.

This is a great start, but should also specify the name mapping itself more formally.

> A name mapping is a list of field mapping objects. Each field mapping has the following properties:
> * `names`: A required list of 0 or more names for a field. Note that names may contain `.`
> * `field-id`: An optional Iceberg field ID to be used for a field with one of the given names
> * `fields`: An optional list of field mappings for child fields of structs, maps, and lists
>
> A field mapping may map multiple names to a single field ID to support cases where a name has been updated. For example, Avro field aliases should also be listed in names. Similarly, fields that exist only in the Iceberg schema may be in the field mapping with an empty list of names, and fields that exist in imported files but not in the Iceberg schema may omit `field-id`.
>
> Mappings for list types should contain a child mapping for the "element" field and mappings for map types should contain child mappings for "key" and "value" fields.
>
> Fields that are not mapped to IDs must be ignored.

Reply:
Sounds good to me, i'll make the mods

Reply:
Oh, and if name does contain `.`, then the field name itself must contain `.`. A mapping for `names: ["a.b"]` will map a field called `"a.b"` and does NOT apply to field `"b"` nested within a field `"a"`.

Reply:
What do you think about including an example JSON? I was thinking of just adding in the example in NameMappingParser.java

Reply:
I think we should have a section on serializing these in the appendix and can give examples there, like the others. Does that work?

Reply:
I think it's easier to have whole examples, but I added one in the style of the other appendixes

Reply:
I agree that a whole example would be rather helpful